_include/q/_utils.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+SCRIPT_DIR=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)
+CONFIG_DIR="${SCRIPT_DIR}/../../docs/config.toml"
+
+export TOP_PID=$$
+trap 'exit 1' TERM
+
+function extract_parameter() {
+  if [[ "$#" != 1 ]]; then
+    echo "Fatal error: No parameter or too many parameter passed: $@" >&2
+    exit_script
+  fi
+  parameter_value=`awk -F'"' '/'$1'[ ]*=[ ]*"/{print $2}' $CONFIG_DIR`
+  if [ "$parameter_value" = "" ]; then
+    echo "Fatal error: $1 parameter no found valid value." >&2
+    exit_script
+  fi
+  echo ${parameter_value}
+}
+
+function exit_script() {
```

Comment:
```suggestion
function trigger_fatal_error() {
  echo $1 >&2
```
nit: we could move the error message into the fatal error function to reduce code redundancy even more. ...just as an idea. You don't have to go for it if you think it's too much. The script is small enough to grasp what it's doing, anyway.